pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -199,18 +207,43 @@ func convertRbacRulesToFilterConfig(service string, roles []model.Config, bindin
 				permissions = append(permissions, convertToPermission(rule))
 			}
 		}
+		if len(permissions) == 0 {
+			log.Debugf("role %v skipped for no permissions found", role.Name)
+			continue
+		}
 
-		if len(permissions) != 0 {
+		// If ServiceRole is in PERMISSIVE mode, all ServiceRoleBindings that refers to it will be treated as PERMISSIVE mode.
+		if role.Spec.(*rbacproto.ServiceRole).Mode == rbacproto.EnforcementMode_PERMISSIVE {
```

Comment:
you shouldn't need this if, if convertToPrincipal already take care of it.

Reply:
push the logic down to convertToPrincipals